src/item.cpp

```diff
@@ -1706,6 +1706,19 @@ void item::on_wield( player &p  )
     if( &p == &g->u && art != nullptr ) {
         g->add_artifact_messages( art->effects_wielded );
     }
+
+    if (is_gun() && has_gunmod("folding_stock") != -1) {
```

Comment:
Instead of checking for the specific item, please add a flag like "SLOW_WIELD" to the stock and check for that instead.